### ISSUE TITLE
Networking: Net table indexes receive negative values and crash the game

### ISSUE THREAD

**Issue by DevulTj** [Facepunch/sbox-issues#1358]
**Describe the bug**
Net table indexes receive negative values and crash the game

**To Reproduce**

1. Download my repro case at https://github.com/DevulTj/sbox-netlab
2. Join a match
3. Watch the game state change, and crash eventually

![ShareX_XMGWdAIR9G](https://user-images.githubusercontent.com/22372616/144664033-e50d1ebf-fafb-47c9-9af2-b671f42e18b8.png)
![sbox_xx8p8z9cCT](https://user-images.githubusercontent.com/22372616/144664200-ae142797-d56e-43d0-a5d2-f3912ffac028.png)
![devenv_wRsgzp137s](https://user-images.githubusercontent.com/22372616/144665663-f2962383-4a83-454a-9ce0-1695da07fcfd.png)

**Comment by handsomematt** [comment]
This was a rabbit hole, should be working now. I had to change quite a few things so let me know if that's broken anything else, I tested a lot of games and they all seemed fine.

Your example boiled down to the following faulty code:

```csharp
[Net] public DataClass OldDataClass { get; set; }
[Net] public DataClass DataClass { get; set; }

OldDataClass = DataClass;
DataClass = new DataClass();
```

Where OldDataClass was being set to DataClass, but then the reassignment of DataClass was releasing the variables making the network table invalid.
